tools/rosmsg/src/rosmsg/__init__.py

```diff
@@ -683,8 +683,16 @@ def rosmsg_cmd_packages(mode, full, argv=None):
 def rosmsg_cmd_list(mode, full, argv=None):
     if argv is None:
         argv = sys.argv[1:]
-    parser = OptionParser(usage="usage: ros%s list"%mode[1:])
+    parser = OptionParser(usage="usage: ros%s list [package_name]"%mode[1:])
     options, args = parser.parse_args(argv[1:])
+    if len(args) == 0:
+        package_name = None
+    elif len(args) == 1:
+        package_name = args[0]
```

Comment:
Since this is using `OptionParser` can you please make use of `add_option` to actually extract the passed option (rather then bypassing it and using `args[0]`. You could also update the code to `argparse` if you like.